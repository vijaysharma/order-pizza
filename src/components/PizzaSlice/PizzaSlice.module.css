.PizzaSlice {
  text-align: right;
  padding: 1rem;
}
@media only screen and (max-width: 640px) {
  .PizzaSlice {
    padding: .75rem;
  }
}