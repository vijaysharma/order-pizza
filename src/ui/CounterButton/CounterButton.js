import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faMinusCircle, faPlusCircle} from '@fortawesome/free-solid-svg-icons';
import * as classes from './CounterButton.module.css';

const CounterButton = props => (
  <div className={classes.CounterButton}>
    <FontAwesomeIcon icon={faMinusCircle} onClick={props.decrement} className={[props.disabled ? classes.DecrementDisabled : '', classes.Decrement].join(' ')} />
    <span>{props.count}</span>
    <FontAwesomeIcon icon={faPlusCircle} onClick={props.increment} className={classes.Increment} />
  </div>
);


export default CounterButton;