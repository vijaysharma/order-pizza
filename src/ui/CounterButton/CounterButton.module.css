.CounterButton {
  color: #9b9b9f;
  display: inline-block;
}
.CounterButton svg {
  vertical-align: middle;
}
.CounterButton span {
  font-size: 1rem;
  padding: 0 0.75rem;
  margin: 0 0.75rem;
  display: inline-block;
  vertical-align: middle;
  width: 2.5rem;
  text-align: center;
}
.Increment, .Decrement {
  cursor: pointer;
  font-size: 2rem;
}
.Increment {
  color: #e13b5b;
}
.Decrement {
  color: #283688;
}
.DecrementDisabled {
  color: #8b8b8b;
  cursor: not-allowed;
}