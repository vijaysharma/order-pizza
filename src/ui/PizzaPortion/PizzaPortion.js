import React from 'react';
import PropTypes from 'prop-types';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faPizzaSlice} from '@fortawesome/free-solid-svg-icons';

import * as classes from './PizzaPortion.module.css';

const PizzaPortion = props => {
  let sliceClass = '';
  switch (props.size) {
    case 'small':
      sliceClass = classes.SmallSlice;
      break;
    case 'medium':
      sliceClass = classes.MediumSlice;
      break;
    case 'large':
      sliceClass = classes.LargeSlice;
      break;
    default:
      sliceClass = '';
  }
  const label = props.label ? props.label.toUpperCase() : '';
  return (
    <div className={classes.PizzaSlice}>
      <FontAwesomeIcon icon={faPizzaSlice} className={sliceClass} />
      <span>{label}</span>
    </div>
  );
};
PizzaPortion.defaultProps = {
  size: 'small',
  label: 'small'
};
PizzaPortion.propTypes = {
  size: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};
export default PizzaPortion;