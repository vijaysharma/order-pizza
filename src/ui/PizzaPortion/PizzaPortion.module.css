.PizzaSlice {
  color: #283688;
  font-size: 1rem;
  display: inline-block;
}

.PizzaSlice span {
  display: inline-block;
  margin: 0 2.5rem;
  width: 5rem;
  vertical-align: middle;
  text-align: left;
}

.SmallSlice, .MediumSlice, .LargeSlice {
  vertical-align: middle;
}

.PizzaSlice .SmallSlice, .PizzaSlice .MediumSlice, .PizzaSlice .LargeSlice {
  width: 1.5rem;
}

.SmallSlice {
  font-size: 1.25rem;
}

.MediumSlice {
  font-size: 1.5rem;
}

.LargeSlice {
  font-size: 1.75rem;
}