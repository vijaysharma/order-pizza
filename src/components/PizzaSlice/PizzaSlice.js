import React from 'react';
import PizzaPortion from '../../ui/PizzaPortion/PizzaPortion';
import CounterButton from '../../ui/CounterButton/CounterButton';
import * as classes from './PizzaSlice.module.css';

const PizzaSlice = props => {
  return (
    <div className={classes.PizzaSlice}>
      <PizzaPortion
        size={props.size}
        label={props.label} />
      <CounterButton
        count={props.count}
        increment={props.onIncrement}
        decrement={props.onDecrement}
        disabled={props.disabled} />
    </div>
  );
};

export default PizzaSlice;