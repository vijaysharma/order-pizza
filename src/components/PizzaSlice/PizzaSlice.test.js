import React from 'react';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PizzaSlice from './PizzaSlice';
import PizzaPortion from '../../ui/PizzaPortion/PizzaPortion';
import CounterButton from '../../ui/CounterButton/CounterButton';

configure({adapter: new Adapter()});

describe('<PizzaSlice/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PizzaSlice />);
  });

  it('should render <PizzaSlice/> component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render <PizzaSlice/> correctly', () => {
    wrapper.setProps({
      label: 'medium',
      size: 'medium'
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <PizzaSlice/> component', () => {
    expect(wrapper.find(CounterButton)).toHaveLength(1);
  });

  it('should render <PizzaSlice/> component', () => {
    expect(wrapper.find(PizzaPortion)).toHaveLength(1);
  });
});