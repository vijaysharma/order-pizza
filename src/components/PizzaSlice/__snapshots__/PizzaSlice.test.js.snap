// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PizzaSlice/> should render <PizzaSlice/> correctly 1`] = `ShallowWrapper {}`;
